GVFS/GVFS/CommandLine/SparseVerb.cs

```diff
@@ -599,45 +596,52 @@ private void CheckGitStatus(ITracer tracer, GVFSEnlistment enlistment, HashSet<s
                 }
                 else
                 {
-                    this.WriteMessage(tracer, statusResult.Output);
-                    this.WriteMessage(tracer, "git status reported that you have dirty files");
-                    this.WriteMessage(tracer, "Either commit your changes or reset and clean");
+                    this.WriteMessage(tracer, "git status reported that you have dirty files:\n");
+                    foreach (string path in dirtyUncoveredPaths)
+                    {
+                        this.WriteMessage(tracer, $"    {path}");
+                    }
+
+                    this.WriteMessage(tracer, "\nEither commit your changes or reset and clean");
                 }
 
                 this.Output.WriteLine();
                 this.ReportErrorAndExit(tracer, "Sparse was aborted.");
             }
         }
 
-        private bool ContainsPathNotCoveredBySparseFolders(string statusOutput, HashSet<string> sparseFolders)
+        private HashSet<string> GetPathsNotCoveredBySparseFolders(string statusOutput, HashSet<string> sparseFolders)

```

Comment:
I was thinking maybe we could gather the paths from git status output and then use Linq's `Except`?  I'm not sure how much if any it would simplify this code.

Reply:
> I'm not sure how much if any it would simplify this code.

Yeah I think in this case it might make it a bit more complicated because of the rename logic (having to track what paths are for a rename) and so I'm thinking it's simpler to stick with the current approach.